src/deferred.js

```diff
@@ -182,6 +187,12 @@ jQuery.extend( {
 							if ( depth ) {
 								process();
 							} else {
+
+								// Call an optional hook to record the stack, in case of exception
+								// since it's otherwise lost when execution goes async
+								if ( jQuery.Deferred.getStackHook ) {
+									process.stackTrace = jQuery.Deferred.getStackHook();
```

Comment:
The problem is that we have to get the stack before the `setTimeout` and at that point we don't know whether there will be a problem or not. If you run the tests with the console.trace() there right now you can see how noisy that would be. We only want to dump the stack if there's an exception.